### PR TITLE
Make '@backstage/core-app-api' a dev dep on all plugins

### PR DESCRIPTION
**Pull request by nicolasarnold12321**
#### :heavy_check_mark: Checklist

- [ ] Added tests for new functionality and regression tests for bug fixes
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

**Comment by Xantier** [comment]
The dependency is used in 3 places on the application code. You need to modify those to work in a different way.
`plugins/frontend/backstage-plugin-github-pull-requests/src/components/Router.tsx`
`plugins/frontend/backstage-plugin-travis-ci/src/Router.tsx`
`plugins/frontend/backstage-plugin-buildkite/src/components/Router.tsx`

**Review by Irma12 — DISMISSED** [review]

**Comment by benjdlambert** [comment]
I believe that you can just use a `Router` from `react-router` in place of `FlatRoutes` for your usages here.

File: plugins/frontend/backstage-plugin-github-insights/src/mocks/scmIntegrationsApiMock.test.ts [hunk 1]
```diff
@@ -45,3 +45,10 @@ export const createScmIntegrationsApiMock = (
     ]),
   );
 };
+
+
+describe('github-insights', () => {
```
Comment (Xantier) [inline plugins/frontend/backstage-plugin-github-insights/src/mocks/scmIntegrationsApiMock.test.ts:50]: lol. I'll allow it.

**Review by Xantier — APPROVED** [review]

File: plugins/frontend/backstage-plugin-github-insights/src/mocks/scmIntegrationsApiMock.test.ts (hunk 1 above)
Comment (Xantier) [reply plugins/frontend/backstage-plugin-github-insights/src/mocks/scmIntegrationsApiMock.test.ts:50]: ah, linter doesn't. You can export a function that gets called and you can pass the offending dependency into that function from an actual test file.

**Review by Xantier — APPROVED** [review]

**Pull request merged**